part-5/db_extraction-physiological-parameters.md

```diff
@@ -0,0 +1,164 @@
+db <- readWorkbook(file.path)
+df <- tibble::as_tibble(db)
+
+# Transform parameter names to fit OSPS parameter paths
+df <- df %>% mutate(concatenated_path = paste(ContainerName, ParameterName, sep = "|"))
+
+# Find distributed parameters
+distributed_parameters <- df %>%
+  dplyr::filter(Age >= 30, Age <= 31) %>%
+  dplyr::group_by(concatenated_path) %>%
+  dplyr::filter(n() > 1) %>%
+  dplyr::pull(concatenated_path) %>%
+  unique()
+
+# Isolate distributed parameters & save to worksheet
+distributed_parameters_df <- df %>%
+  dplyr::filter(concatenated_path  %in% distributed_parameters) %>%
+  dplyr::select(Age, concatenated_path, Mean)
+  
+# Write data in workbook
+wb <- createWorkbook()
+addWorksheet(wb, "Postpartum_distributedParam")
+
+writeData(wb, sheet = "Postpartum_distributedParam", 
+          distributed_parameters_df)
+
+saveWorkbook(wb, "data/Postpartum_distributed_parameters.xlsx", overwrite = TRUE) 
+```
+
+The user interface of the database viewer tool (e.g. Db Browser for SQLite) may also be used to extract the parameters of interest (i.e. distributed parameters).
+
+### Import in Mobi as Table & create time-varying parameters
+
+In Mobi:
+
+1.  Create a Postpartum_Age parameter increasing over simulation time (NB: the parameter 'Age' remains constant):
+
+    -   Define the value for the age parameter at the beginning of the simulation 'Age':
+
+        -   Formula type is "constant (single numeric value)", dimension is 'Age in years', Value is 30 (i.e. start age of the postpartum population in database)
+
+    -   Add parameter 'Organism\|Postpartum Age'
+
+        -   Dimension is Time (default unit is minutes, in OSPS)
+
+        -   Formula type is 'Formula (an explicit formula)'
+
+        -   Formula is "Age \*year2min + TIME" (the start Age in years must be converted into the OSPS default unit for 'TIME', which is minute)
+
+        ![Fig1: Postpartum Age](../assets/images/part-5/Postpartum_age.png){width="300"}
+
+2.  Open organ compartment that will contain the time-varying parameter (e.g. Breasts)
+
+3.  Add an intermediary Table parameter (E.g. Volume_breasts_TABLE)
+
+    -   Chose the formula type "Table (multiple time discrete and piecewise constant numerical values)
+
+    -   Define a formula name (e.g. TABLE_volume_Breasts_PP)
+
+    -   Click on the 'import from worksheet' button (green arrow on Figure below)
+
+    -   Select a parameter from the data file import menu. E.g for the Postpartum case:
+
+        -   Chose "data/Postpartum_distributed_parameters.xlsx" as exported in previous section
+
+        -   Filter 'concatenated_path' \> "Breasts\|Volume"
+
+        -   Enter units (by default values are in the base unit of the Db, see file: [OSPSuite.Dimensions.xml](https://esqlabs.sharepoint.com/:u:/s/S-BASF-P23-195A/EZSeZvDmQFRLvKNCJRqyxyUBkv8jR2po28wDa-caVE1LMg?e=BagOhT)). Column 'Age' in years and column 'Mean' in litter (OSPS base unit for volumes)
```

Comment:
please replace this internal link with a valid OSP link
https://github.com/Open-Systems-Pharmacology/OSPSuite.Dimensions/blob/master/OSPSuite.Dimensions.xml